packages/template-retail-react-app/app/ssr.js

```diff
@@ -432,6 +402,65 @@ const {handler} = runtime.createHandler(options, (app) => {
     app.get('/favicon.ico', runtime.serveStaticFile('static/ico/favicon.ico'))
 
     app.get('/worker.js(.map)?', runtime.serveServiceWorker)
+
+    // Helper function to transform relative icon paths to absolute URLs
+    function transformIconPaths(data, ecomServerHost) {
+        const baseUrl = `https://${ecomServerHost}/on/demandware.static/Sites-Site/-/-/internal`
+        const dataStr = JSON.stringify(data)
+        // Replace all relative icon paths with absolute URLs
+        const transformedStr = dataStr.replace(/"src":\s*"\/icons\//g, `"src":"${baseUrl}/icons/`)
+        return JSON.parse(transformedStr)
+    }
+
+    app.get('/api/payment-metadata', async (req, res) => {
```

Comment:
We don't currently have a pattern for the endpoint path to be `/api/*`. To be more aligned with other existing endpoints, how about without the `api` portion?

```suggestion
    app.get('/payment-metadata', async (req, res) => {
```

Or perhaps if there's a plan for another payment related endpoints, how about `/payments/metadata`?

Reply:
sure, we are open to whichever convention works.  I think that shouldn't break us since its a proxy api path anyway.  
/payments/metadata sounds reasonable.  We just have to modify a few test classes and a query logic we had in place.  

 const {data: serverMetadata, isLoading: serverMetadataLoading} = useQuery({
        queryKey: ['payment-metadata'],
        queryFn: async () => {
            const response = await fetch(`${appOrigin}/api/payment-metadata`)